coralproject/talk
# [CORL-2562] During sign up or change username, make sure username is unique

## What does this PR do?

When signing up or changing your username, check that the proposed username is a unique username and not used by anyone else.

Notifies the user with an error if the username is already taken.

## What changes to the GraphQL/Database Schema does this PR introduce?

None

## Does this PR introduce any new environment variables or feature flags? 

No

## If any indexes were added, were they added to `INDEXES.md`?

No new indexes

## How do I test this PR?

* try and sign up (local auth) and choose a username that is already taken
   * see that it blocks you from creating the account and notifies you the username is taken
* try and sign up (local auth) and choose a new username that is NOT taken
   * see that you are able to successfully create the new unique user
* enable username changes in Admin via `Configure > Authentication > Commenter account management features > Allow users to > Change their usernames`
* attempt to change username to a name that is already taken
   * see that it blocks you from changing username and notifies you the username is taken
* change the username to something that is NOT taken
   * see that it allows you change to a valid, non-taken username
 
## How do we deploy this PR?

No special considerations
